### ISSUE TITLE
[enhancement] Custom backgrounds and theming

### ISSUE THREAD

**Issue by ShadowJonathan** [SchildiChat/schildichat-desktop#34]
(Note: I have seen this float around in the labs functionality, not sure if this lands here or in upstream element)

**Comment by benneti** [comment]
Using these themes https://github.com/aaronraimist/element-themes seems to work just fine.
Is it possible to specify the bubble color in a theme.json file?

**Comment by su-ex** [comment]
@benneti 
Good to hear it's not broken. I think bubble color could be added.
What are the steps required and what's the fastest way to test a custom theme (at best with config only).

**Comment by benneti** [comment]
actually I think I need, in theory it should be possible using a config.json
``` json
{
    "settingDefaults": {
        "custom_themes": [
            {
                "name": "Dracula BT",
                "is_dark": true,
                "colors": {
                    "accent-color": "#bd93f9",
                    "primary-color": "#bd93f9",
                    "warning-color": "#ff5555",
                    "sidebar-color": "#1e1f29",
                    "roomlist-background-color": "#1e1f29",
                    "roomlist-text-color": "#f8f8f2",
                    "roomlist-text-secondary-color": "#f8f8f2",
                    "roomlist-highlights-color": "#00000030",
                    "roomlist-separator-color": "#4d4d4d90",
                    "timeline-background-color": "#282A36",
                    "timeline-text-color": "#f8f8f2",
                    "timeline-text-secondary-color": "#dddddd",
                    "timeline-highlights-color": "#00000030",
                    "reaction-row-button-selected-bg-color": "#b9bbbe"
                }
            }
        ]
    }
}
```
but if I try it like this it is ignored (in .config/SchildiChat) but I can add it via URL and then select it...
What puzzles me the most is I do not understand at all where these settings are stored (I am sure not in .config/SchildiChat because I deleted this directory and the installed themes were there again after I logged in again.

**Comment by su-ex** [comment]
What do you mean with installed themes? From where to where?
I already tried it via the config like above but this somehow broke my whole config although the json seemed to be valid.

**Comment by benneti** [comment]
If you enable adding cutom themes in the lab, it is possible to add a theme link like https://raw.githubusercontent.com/aaronraimist/element-themes/master/Dracula/Non-flat/Dracula.json in the appearance settings.
Are there any places apart from .config/SchildiChat where configuration files are stores -- I would like to delete them and try to create a config with a custom theme, but for me the config.json seems to be ignored and the settings from somewhere else are used (I also have no Element * folders in .config)?

**Comment by su-ex** [comment]
Got it. I already suspected this but wasn't entirely sure what you meant with installation.
About config paths: https://github.com/SchildiChat/element-desktop#user-specified-configjson
That's what I meant: Once I included a custom theme in my config, the whole config was ignored and it falls back to the hardcoded defaults I assume.

**Comment by benneti** [comment]
Ah, I wonder why this happens and where the hardcoded or in app chosen settings are saved. As some of them persist after deleting .config/SchildiChat (and are not set back to the default)

**Comment by su-ex** [comment]
The default default (intentionally doubled) settings are stored in the sourcecode.

**Comment by benneti** [comment]
Yeah I do not mean them, but is there another location (outside .config where settings are stored).
Because even after deleting my .config/SchildiChat and then logging in again, I have the dracula theme as an option (which is not in the sourcecode)

**Comment by su-ex** [comment]
Does it disappear after logout/login? If yes it's stored in the device data probably. If it exists across multiple devices it's synced (then you can watch the network traffic via devtools).

**Comment by benneti** [comment]
Ah I did not think of this.
This would make it a bit harder to delete the option again.

**Comment by benneti** [comment]
figured out what needs to be done to respect the config file.
The simple thing is to go into a chat advanced and then dev tools, and delete the custom_themes array in im.vector.web.settings then the local config is respected; I did not however figure out how to change the used theme in the config file or where it is saved online...

**Comment by su-ex** [comment]
It's not saved online, the theme seems to be set on the device level: https://github.com/SchildiChat/matrix-react-sdk/blob/23097347d497e0c848b3b43c12c4117e40a71a95/src/components/views/settings/tabs/user/AppearanceUserSettingsTab.tsx#L131

It's probably possible to tamper with the theme inside the config through a combination of [these settings](https://github.com/SchildiChat/matrix-react-sdk/blob/23097347d497e0c848b3b43c12c4117e40a71a95/src/settings/Settings.ts#L377-390) inside the `settingDefaults` key.

**Comment by benneti** [comment]
Yes, I have this set, but I think the setting I applied in the guy overrides this; It now works anyhow the only grain of salt is that to get the setting in config.json applied a fresh login is necessary.
This is the relevant part of my configuration
```
{
  "settingDefaults": {
    "custom_themes": [
      {
        "name": "Dracula",
        "is_dark": true,
        "fonts": {
          "faces": [
            {
              "font-family": "Fira Sans"
            }
          ],
          "general": "Fira Sans",
          "monospace": "Fira Code"
        },
        "colors": {
          "accent-color": "#bd93f9",
          "primary-color": "#bd93f9",
          "warning-color": "#ff5555",
          "sidebar-color": "#1e1f29",
          "roomlist-background-color": "#1e1f29",
          "roomlist-text-color": "#f8f8f2",
          "roomlist-text-secondary-color": "#f8f8f2",
          "roomlist-highlights-color": "#00000030",
          "roomlist-separator-color": "#4d4d4d90",
          "timeline-background-color": "#282A36",
          "timeline-text-color": "#f8f8f2",
          "timeline-text-secondary-color": "#dddddd",
          "timeline-highlights-color": "#00000030",
          "reaction-row-button-selected-bg-color": "#b9bbbe"
        }
      }
    ],
    "breadcrumbs": true,
    "useCustomFontSize": true,
    "theme": "custom-Dracula"
  },
  "default_theme": "custom-Dracula"
}
```
Only thing missing IMHO is setting the message bubble colors (I would like my bubbles colored)

**Comment by su-ex** [comment]
Yes, that's too be expected: A user selection will always overwrite the default values.
I'll include options for the bubble color in the next release.

**Comment by benneti** [comment]
Amazing, thank you!

**Comment by su-ex** [comment]
Custom themes verified working and now with custom color option: https://github.com/SchildiChat/element-web/commit/84434c6e1662114268bd50847f2a57800e45f44f#diff-c97f6c85ec7877988cd739449cbd1ebc45d262faf97e9d6084186744fcdb95e1 ☑️

**Comment by benneti** [comment]
can confirm, works like a charm!

**Comment by Domoel** [comment]
I did everything explanined here (deleting the themes from im.vector.web.settings and putting in all necessary config parameter for config.jason) but the theme does not load after inital login into the app. What am I doing wrong here?

![Image](https://github.com/user-attachments/assets/12be43d2-0266-4e33-b531-7e76ef693196)
